tests/test_model.py

```diff
@@ -198,3 +170,176 @@ def test_nearby(self):
+
+        # The Brooklyn Public Library serves New York City.
+        brooklyn = self._library(
+            "Brooklyn Public Library", [self.new_york_city]
+        )
+
+        # We can find the library by its name.
+        eq_([brooklyn], search("brooklyn public library"))
+
+        # We can tolerate a small number of typos in a name or alias
+        # that is longer than 6 characters.
+        eq_([brooklyn], search("broklyn public library"))
+        get_one_or_create(
+            self._db, LibraryAlias, name="Bklynlib", language=None,
+            library=brooklyn
+        )        
+        eq_([brooklyn], search("zklynlib"))
+
+        # The Boston Public Library serves Boston, MA.
+        boston = self._library(
+            "Boston Public Library", [self.boston_ma]
+        )
+
+        # Both libraries are known colloquially as 'BPL'.
+        for library in (brooklyn, boston):
+            get_one_or_create(
+                self._db, LibraryAlias, name="BPL", language=None,
+                library=library
+            )
+        eq_(
+            set([brooklyn, boston]), set(search("bpl"))
+        )
+
+        # We do not tolerate typos in short names, because the chance of
+        # ambiguity is so high.
+        eq_([], search("opl"))
+
+        # If we're searching for "BPL" from California, Brooklyn shows
+        # up first, because it's closer to California.
+        eq_(["Brooklyn Public Library",
+             "Boston Public Library"],
+            [x.name for x in search("bpl", GeometryUtility.point(35, -118))])
+
+        # If we're searching for "BPL" from Maine, Boston shows
+        # up first, because it's closer to Maine.
+        eq_(["Boston Public Library",
+             "Brooklyn Public Library"],
+            [x.name for x in search("bpl", GeometryUtility.point(43, -70))]
+        )
+        
+
+    def test_search_by_location(self):
+        # We know about three libraries.
+        nypl = self.nypl
+        kansas_state = self.kansas_state_library
+        connecticut_state = self.connecticut_state_library
+
+        # The NYPL explicitly covers New York City, which has
+        # 'Manhattan' as an alias.
+        [nyc] = [x.place for x in nypl.service_areas]
+        assert "Manhattan" in [x.name for x in nyc.aliases]
+        
+        # The Kansas state library covers the entire state,
+        # which happens to contain a city called Manhattan.
+        [kansas] = [x.place for x in kansas_state.service_areas]
+        eq_("Kansas", kansas.external_name)
+        eq_(Place.STATE, kansas.type)
+        manhattan_ks = self.manhattan_ks
+        
+        # A search for 'manhattan' finds both libraries.
+        libraries = list(Library.search_by_location_name(self._db, "manhattan"))
+        eq_(set(["NYPL", "Kansas State Library"]),
+            set([x.name for x in libraries])
+        )
+
+        # If you're searching from California, the Kansas library
+        # shows up first.
```

Comment:
I don't think we should do anything about this now, but for this particular example, it would be better for NYPL to show up first.